File: src/main/java/shape/Square.java
package shape;

public class Square{
    private Point pointOne;
    private Point pointTwo;

    public Square(Point pointOne) {
        this.pointOne = pointOne;
        
        //Set point two to opposite corner of the square by cloning point one and moving it x and y values to be the opposite corner of the square
        this.pointTwo = pointOne.Clone();
        this.pointTwo.MovePoint(pointOne.GetX(), -pointOne.GetY());

        
    }

    public Point GetPointOne() {
        return pointOne;
    }
    public Point GetPointTwo() {
        return pointTwo;
    }
    public double GetArea() {
        double area = Math.abs(pointTwo.GetX() - pointOne.GetX()) * Math.abs(pointTwo.GetY() - pointOne.GetY());
        return area;
    }

    public void MoveSquare(double addXvalue, double addYvalue) {
        pointOne.MovePoint(addXvalue, addYvalue);
        pointTwo.MovePoint(addXvalue, addYvalue);
    }
}

File: src/main/kotlin/shape/Square.kt
package shape

import kotlin.math.abs

class Square(private val pointOne: Point) {

    private val pointTwo: Point

    init {
        // Set pointTwo to opposite corner by cloning pointOne
        pointTwo = pointOne.Clone()
        pointTwo.MovePoint(pointOne.GetX(), -pointOne.GetY())

        // Optional: reject 0 area square
        if (GetArea() <= 0) {
            throw IllegalArgumentException("Area cannot be 0")
        }
    }

    fun GetPointOne(): Point {
        return pointOne
    }

    fun GetPointTwo(): Point {
        return pointTwo
    }

    fun GetArea(): Double {
        val area = abs(pointTwo.GetX() - pointOne.GetX()) *
                   abs(pointTwo.GetY() - pointOne.GetY())
        return area
    }

    fun MoveSquare(addXvalue: Double, addYvalue: Double) {
        pointOne.MovePoint(addXvalue, addYvalue)
        pointTwo.MovePoint(addXvalue, addYvalue)
    }
}